src/Microsoft.AspNetCore.SignalR/TypedClientBuilder.cs

```diff
@@ -0,0 +1,243 @@
+
+            if (genericTypeNames.Any())
+            {
+                methodBuilder.DefineGenericParameters(genericTypeNames);
+            }
+
+            ILGenerator generator = methodBuilder.GetILGenerator();
+
+            // Declare local variable to store the arguments to IClientProxy.Invoke
+            generator.DeclareLocal(typeof(object[]));
+
+            // Get IClientProxy
+            generator.Emit(OpCodes.Ldarg_0);
+            generator.Emit(OpCodes.Ldfld, proxyField);
+
+            // The first argument to IClientProxy.Invoke is this method's name
+            generator.Emit(OpCodes.Ldstr, interfaceMethodInfo.Name);
+
+            // Create an new object array to hold all the parameters to this method
+            generator.Emit(OpCodes.Ldc_I4, parameters.Length);
+            generator.Emit(OpCodes.Newarr, typeof(object));
+            generator.Emit(OpCodes.Stloc_0);
+
+            // Store each parameter in the object array
+            for (int i = 0; i < paramTypes.Length; i++)
+            {
+                generator.Emit(OpCodes.Ldloc_0);
+                generator.Emit(OpCodes.Ldc_I4, i);
+                generator.Emit(OpCodes.Ldarg, i + 1);
+                generator.Emit(OpCodes.Box, paramTypes[i]);
+                generator.Emit(OpCodes.Stelem_Ref);
+            }
+
+            // Call IProxy.Invoke
+            generator.Emit(OpCodes.Ldloc_0);
+            generator.Emit(OpCodes.Callvirt, invokeMethod);
+
+            if (interfaceMethodInfo.ReturnType == typeof(void))
+            {
+                // void return
+                generator.Emit(OpCodes.Pop);
+            }
+
+            generator.Emit(OpCodes.Ret);
+        }
+
+        private static void VerifyInterface(Type interfaceType)
+        {
+            if (!interfaceType.IsInterface)
+            {
+                throw new InvalidOperationException("Type must be interface");
+                    //String.Format(CultureInfo.CurrentCulture, Resources.Error_TypeMustBeInterface, interfaceType.Name));
+            }
+
+            if (interfaceType.GetProperties().Length != 0)
+            {
+                throw new InvalidOperationException("Type must not contain properties");
+                    //String.Format(CultureInfo.CurrentCulture, Resources.Error_TypeMustNotContainProperties, interfaceType.Name));
+            }
+
+            if (interfaceType.GetEvents().Length != 0)
+            {
+                throw new InvalidOperationException("Type can't contain events");
+                    //String.Format(CultureInfo.CurrentCulture, Resources.Error_TypeMustNotContainEvents, interfaceType.Name));
+            }
+
+            foreach (var method in interfaceType.GetMethods())
+            {
+                VerifyMethod(interfaceType, method);
+            }
+
+            foreach (var parent in interfaceType.GetInterfaces())
+            {
+                VerifyInterface(parent);
```

Comment:
I'm not too familiar with who controls interface generation, but you may need to check if you have already visited this interface as this could cause infinite recursion.